File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
     rev = 0
     n = x

     if n < 0:
        return False

     while n > 0:
        d = n % 10
        rev = rev * 10 + d
        n //= 10 
     return rev == x

File: 9-palindrome-number/palindrome-number.go
func isPalindrome(x int) bool {

    if x < 0 {
        return false
    }

    var builder strings.Builder
    num := strconv.Itoa(x)
    runes := []rune (num)
    for i := len(runes) - 1; i >= 0; i-- {
        builder.WriteString(string(runes[i]))
    }

    res := builder.String()
   return res == num
}